ioservice/io_foms.c

```diff
@@ -2321,6 +2321,33 @@ static int m0_io_fom_cob_rw_tick(struct m0_fom *fom)
 	M0_ASSERT(rc == M0_FSO_AGAIN || rc == M0_FSO_WAIT);
 
 	M0_ASSERT(m0_io_fom_cob_rw_invariant(fom_obj));
+
+	if (m0_fom_phase(fom) == M0_FOPH_SUCCESS &&
+	    m0_is_write_fop(fom->fo_fop)) {
+		int                 bc_rc;
+		uint64_t            old_cob_size;
+		struct m0_cob_bckey key;
+
+		key.cbk_pfid = pver;
+		key.cbk_user_id = M0_BYTECOUNT_USER_ID;
+		bc_rc = fom_cob_locate(fom);
+		if (bc_rc == 0) {
+			cob = fom_obj->fcrw_cob;
+			old_cob_size = cob->co_nsrec.cnr_size;
+			cob_bytecount_increment(cob, &key, byte_count,
+						m0_fom_tx(fom));
+			/**
+			 * XXX: Overlapping cob extentds are not accounted for
+			 * during cob overwrite. IF a cob is overwritten,
+			 * it will make cob size inaccurate.
+			 */
+			bc_rc = m0_cob_size_update(cob, old_cob_size + byte_count,
+						   m0_fom_tx(fom));
+			if (bc_rc != 0)
+				M0_ERR_INFO(bc_rc, "Failed to update cob_size");
+		}
```

Comment:
Let's fail the fom if cob size update fails, move this code to io_finish(). There will be a loop of stob  io in case of not ecnough net buffers are available.

Reply:
Done.